### PR TITLE
feat: add list api protos

### PR DESCRIPTION
**Pull request by anp13**

### CODE REVIEW

**Review by kvcache — APPROVED** [review]
lgtm

File: proto/cacheclient.proto [hunk 1]
```diff
@@ -175,3 +185,105 @@ message _SetDifferenceResponse {
...
+message _ListPushBackResponse {}
+
+message _ListPopFrontRequest {
+  bytes list_name = 1;
+}
+
+message _ListPopFrontResponse {
+  message _Found {
+    bytes front = 1;
+  }
```
Comment (malandis) [inline proto/cacheclient.proto:216]: Can we easily change this to support a request to pop `count` items from a list `list_name`?

File: proto/cacheclient.proto [hunk 2]
```diff
@@ -175,3 +185,105 @@ message _SetDifferenceResponse {
...
+
+message _ListPopBackRequest {
+  bytes list_name = 1;
+}
+
+message _ListPopBackResponse {
+  message _Found {
+    bytes back = 1;
+  }
+
```
Comment (malandis) [inline proto/cacheclient.proto:234]: Same question as for ``ListPopFront`

File: proto/cacheclient.proto (hunk 1 above)
Comment (anp13) [reply proto/cacheclient.proto:216]: Good callout. @kvcache we should make the change on the blobdb server as well.

Comment (anp13) [reply proto/cacheclient.proto:229]: @malandis going to leave this as is for now, should be easy to add this functionality later.

**Review by malandis — APPROVED** [review]

**Pull request merged**